Tasks/AzureAppServiceSettingsV1/azurewebappsettings.ts

```diff
@@ -0,0 +1,77 @@
+import tl = require('azure-pipelines-task-lib/task');
+import path = require('path');
+import { AzureEndpoint } from 'azurermdeploycommon/azure-arm-rest/azureModels';
+import { AzureRMEndpoint, dispose } from 'azurermdeploycommon/azure-arm-rest/azure-arm-endpoint';
+import { AzureAppService } from 'azurermdeploycommon/azure-arm-rest/azure-arm-app-service';
+import { AzureAppServiceUtility } from 'azurermdeploycommon/operations/AzureAppServiceUtility';
+import { Resources } from 'azurermdeploycommon/azure-arm-rest/azure-arm-resource';
+
+export class AzureResourceFilterUtils {
+    public static async getResourceGroupName(endpoint: AzureEndpoint, resourceType: string, resourceName: string): Promise<string> {
+        var azureResources: Resources = new Resources(endpoint);
+        var filteredResources: Array<any> = await azureResources.getResources(resourceType, resourceName);
+        let resourceGroupName: string;
+        if(!filteredResources || filteredResources.length == 0) {
+            throw new Error(tl.loc('ResourceDoesntExist', resourceName));
+        }
+        else if(filteredResources.length == 1) {
+            resourceGroupName = filteredResources[0].id.split("/")[4];
+        }
+        else {
+            throw new Error(tl.loc('MultipleResourceGroupFoundForAppService', resourceName));
+        }
+
+        return resourceGroupName;
+    }
+}
+
+
+async function main() {
+
+    try {
+        tl.setResourcePath(path.join( __dirname, 'task.json'));
+        tl.setResourcePath(path.join( __dirname, 'node_modules/azurermdeploycommon/module.json'));
+        var connectedServiceName = tl.getInput('ConnectedServiceName', true);
+        var webAppName: string = tl.getInput('WebAppName', true);
+        var resourceGroupName: string = tl.getInput('ResourceGroupName', false);
+        var slotName: string = tl.getInput('slotName', false);
+        var AppSettings: string = tl.getInput('appSettings', false);
+        var ConfigurationSettings: string = tl.getInput('generalSettings', false);
+        var ConnectionStrings: string = tl.getInput('connectionStrings', false);
+
+        var azureEndpoint: AzureEndpoint = await new AzureRMEndpoint(connectedServiceName).getEndpoint();
+        console.log(tl.loc('GotconnectiondetailsforazureRMWebApp0', webAppName));
+        resourceGroupName = await AzureResourceFilterUtils.getResourceGroupName(azureEndpoint, 'Microsoft.Web/Sites', webAppName);
```

Comment:
query only when resourceGroupname is not provided. We will prevent this call and improve on execution time.